pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -59,66 +60,138 @@ func (dv *openShiftClusterDynamicValidator) Dynamic(ctx context.Context) error {
 	// FP validation
 	fpDynamic, err := dynamic.NewValidator(dv.log, dv.env, dv.env.Environment(), dv.subscriptionDoc.ID, dv.fpAuthorizer, dynamic.AuthorizerFirstParty)
 	if err != nil {
-		return err
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidParameter,
+			"First Principal",
```

Comment:
1. First Principal - should probably be first party principal.
2. But in general, I think passing "first party principal" as `target` is not right. This is supposeted to be a path to a property. This appleis to almost all changes in this PR.

Reply:
Thanks for these pointers, I have made changes to remove the targets and used `First Party Principal` in this error message.